lib/change-layout.jsx

```diff
@@ -0,0 +1,61 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+
+import {
+	showPane as showPaneAction,
+	hidePane as hidePaneAction,
+} from './state/settings/actions';
+import ThreePanesIcon from './icons/three-panes';
+import TwoPanesIcon from './icons/two-panes';
+import EditorOnlyIcon from './icons/editor-only';
+
+class ChangeLayout extends Component {
+
+	changeLayout = () => {
+		const {
+			showPane,
+			hidePane,
+			visiblePanes,
+		} = this.props;
+		if ( visiblePanes.includes( 'navigation-bar' ) ) {
+			hidePane( 'navigation-bar' );
+		} else if ( visiblePanes.includes( 'note-list' ) ) {
+			hidePane( 'note-list' );
+		} else {
+			showPane( 'note-list' );
+			showPane( 'navigation-bar' );
+		}
+	}
+
+	render() {
+		const { visiblePanes } = this.props;
+		return (
+			<button
+				title="Change Layout"
+				className="button button-borderless"
+				onClick={ this.changeLayout }
+			>
+				{ visiblePanes.includes( 'navigation-bar' ) &&
+					<TwoPanesIcon />
+				}
+				{ ! visiblePanes.includes( 'navigation-bar' ) && visiblePanes.includes( 'note-list' ) &&
+					<EditorOnlyIcon />
+				}
+				{ ! visiblePanes.includes( 'note-list' ) && ! visiblePanes.includes( 'navigation-bar' ) &&
+					<ThreePanesIcon />
+				}
+			</button>
```

Comment:
I think this is the behavior suggested by @drw158.
I'm not a fan of it too, but it should be super easy to replace the single icon with all three of them.